src/modals/WalletConnectModal.tsx

```diff
@@ -181,7 +181,7 @@ const WalletConnectModal = ({ onClose }: WalletConnectModalProps) => {
   }
 
   const generateAddressInGroup = () => {
-    const address = generateAddress()
+    const address = generateAddress({ group })
```

Comment:
Somehow we both missed it, just because in the demo it HAPPENED that the next generated address was in group `0`. oopsi